internal/attestation/azure/trustedlaunch/validator.go

```diff
@@ -8,39 +8,98 @@ package trustedlaunch
+
 // Validator for Azure trusted launch VM attestation.
 type Validator struct {
 	oid.AzureTrustedLaunch
 	*vtpm.Validator
+	rootCA []byte
 }
 
 // NewValidator initializes a new Azure validator with the provided PCR values.
 func NewValidator(pcrs map[uint32][]byte, enforcedPCRs []uint32, log vtpm.WarnLogger) *Validator {
-	return &Validator{
-		Validator: vtpm.NewValidator(
-			pcrs,
-			enforcedPCRs,
-			trustedKey,
-			validateVM,
-			vtpm.VerifyPKCS1v15,
-			log,
-		),
-	}
+	v := &Validator{rootCA: []byte(ameRoot)}
+	v.Validator = vtpm.NewValidator(
+		pcrs,
+		enforcedPCRs,
+		v.verifyAttestationKey,
+		validateVM,
+		vtpm.VerifyPKCS1v15,
+		log,
+	)
+	return v
 }
 
-// trustedKey returns the key encoded in the given TPMT_PUBLIC message.
-func trustedKey(akPub, instanceInfo []byte) (crypto.PublicKey, error) {
+// verifyAttestationKey establishes trust in an attestation key.
+// It does so by verifying the certificate chain of the attestation key certificate.
+func (v *Validator) verifyAttestationKey(akPub, instanceInfo []byte) (crypto.PublicKey, error) {
 	pubArea, err := tpm2.DecodePublic(akPub)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("decoding attestation key public area: %w", err)
+	}
+
+	var akSigner akSigner
+	if err := json.Unmarshal(instanceInfo, &akSigner); err != nil {
+		return nil, fmt.Errorf("unmarshaling attestation key signer info: %w", err)
+	}
+
+	akCert, err := x509.ParseCertificate(akSigner.AkCert)
+	if err != nil {
+		return nil, fmt.Errorf("parsing attestation key certificate: %w", err)
+	}
+	akCertCA, err := x509.ParseCertificate(akSigner.CA)
+	if err != nil {
+		return nil, fmt.Errorf("parsing attestation key CA certificate: %w", err)
+	}
+
+	rootCA, err := certutil.PemToX509Cert(v.rootCA)
+	if err != nil {
+		return nil, fmt.Errorf("parsing AME root CA: %w", err)
 	}
-	return pubArea.Key()
+	roots := x509.NewCertPool()
+	roots.AddCert(rootCA)
```

Comment:
This could be done in New or even only once at program start (like global `var ameRootPool = func() *x509.CertPool { ... }()` with panic on error) to catch this logic error early. But not really important, pls do as you prefer.

Reply:
Thought about this too, but wasnt sure if we want to have a potential panic call in the Validator init function.
I guess because the conversion of the root ca from PEM should always work, this it should be ok to panic in case of an error?

Reply:
I guess so. @katexochen once suggested to put such `must` code globally so that it runs (and panics on a programming bug) on program start.